### PR TITLE
v35 - New UI - Account page

### PR DESCRIPTION
**Pull request by DigitalSeneca**
| Screenshot | GIF |
| ------------- | ------------- |
| image | new_ui_settings_page |

### CODE REVIEW

File: lib/pages/settings_pages/components/settings_menu_widget.dart [hunk 1]
```diff
@@ -4,27 +4,32 @@ import 'package:mosquito_alert_app/utils/style.dart';
...
+            Icons.arrow_forward_ios,
+            color: Colors.black,
+            size: 18,
+          ),
+          visualDensity: VisualDensity(vertical: -4),
+          dense: true,          
         ),
-      ),
+        if (addDivider)
+          Container( padding: EdgeInsets.only(left: 10.0, right: 10.0),
```
Comment (epou) [inline lib/pages/settings_pages/components/settings_menu_widget.dart:29]: I was thinking if this padding could be replaced by a combination of `Divider.height`, `Divider.indent` and `Divider.endIndent` parameters

File: lib/pages/settings_pages/settings_page.dart [hunk 2]
```diff
@@ -85,15 +90,25 @@ class _SettingsPageState extends State<SettingsPage> {
...
+    setState((){
+      userScore = _userScore ?? 0;
+    });
+  }
+
   @override
   Widget build(BuildContext context) {
     final key = GlobalKey<ScaffoldState>();
 
     return Scaffold(
```
Comment (epou) [inline lib/pages/settings_pages/settings_page.dart:104]: Nothing special to comment about the code, just being a little persnickety, I suggest double-think if using functions would improve readability and development on your end.

Something like?

```dart
@override
Widget build(BuildContext context) {
  final key = GlobalKey<ScaffoldState>();

  return Scaffold(
    key: key,
    appBar: AppBar(
      title: Text("Account (Hardcoded)"),
    ),
    body: SingleChildScrollView(
      child: Container(
        margin: EdgeInsets.fromLTRB(15, 0, 15, 15),
        child: isLoading
            ? _buildLoadingIndicator(context)
            : _buildAccountContent(context),
      ),
    ),
  );
}

Widget _buildLoadingIndicator(BuildContext context) {
  return Container(
    height: MediaQuery.of(context).size.height * 0.8,
    child: Center(child: CircularProgressIndicator()),
  );
}

Widget _buildAccountContent(BuildContext context) {
  return Column(
    crossAxisAlignment: CrossAxisAlignment.start,
    children: <Widget>[
      _buildProfileHeader(context),
      SizedBox(height: 10),
      Divider(),
      _buildSettingsSection(context),
      _buildSupportSection(context),
      _buildLegalSection(context),
    ],
  );
}

Widget _buildProfileHeader(BuildContext context) {
  return Column(
    children: [
      _buildAvatar(),
      _buildUserName(),
      _buildUserScore(),
    ],
  );
}

Widget _buildAvatar() {
  return Row(
    mainAxisAlignment: MainAxisAlignment.center,
    children: [
      Container(
        decoration: BoxDecoration(
          shape: BoxShape.circle,
          border: Border.all(
            color: Colors.black,
            width: 6,
          ),
        ),
        child: CircleAvatar(
          radius: 25,
          child: Icon(Icons.person, size: 40),
          backgroundColor: Colors.transparent,
        ),
      ),
    ],
  );
}

Widget _buildUserName() {
  return Row(
    mainAxisAlignment: MainAxisAlignment.center,
    children: [
      Text(
        'Anonymous',
        style: TextStyle(fontSize: 24),
      ),
    ],
  );
}

Widget _buildUserScore() {
  return Row(
    mainAxisAlignment: MainAxisAlignment.center,
    children: [
      Icon(
        Icons.star,
        color: Colors.orange,
        size: 15,
      ),
      SizedBox(width: 5),
      Text('$userScore'),
    ],
  );
}

Widget _buildSettingsSection(BuildContext context) {
  return Column(
    crossAxisAlignment: CrossAxisAlignment.start,
    children: [
      Text(MyLocalizations.of(context, "settings_title").toUpperCase()),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'select_language_txt'),
        _openLanguagePickerDialog,
      ),
      _buildBackgroundTrackingSwitch(context),
      _buildAutoTaggingSettings(context),
    ],
  );
}

Widget _buildBackgroundTrackingSwitch(BuildContext context) {
  return SwitchListTile(
    title: Style.body(MyLocalizations.of(context, 'background_tracking_title')),
    subtitle: Padding(
      padding: EdgeInsets.only(top: 8.0),
      child: Text(
        MyLocalizations.of(context, 'background_tracking_subtitle'),
        style: TextStyle(fontSize: 11),
      ),
    ),
    value: isBgTrackingEnabled,
    activeColor: Colors.orange,
    onChanged: (bool value) async {
      await UserManager.setTracking(value);
      var trackingStatus = await UserManager.getTracking();
      setState(() {
        isBgTrackingEnabled = trackingStatus;
      });
      if (!isBgTrackingEnabled) {
        await Workmanager().cancelByTag('trackingTask');
      }
    },
  );
}

Widget _buildAutoTaggingSettings(BuildContext context) {
  return Row(
    children: [
      Expanded(
        child: ExpansionTile(
          initiallyExpanded: numTagsAdded! > 0,
          title: Row(
            children: [
              Text(
                MyLocalizations.of(context, 'auto_tagging_settings_title'),
                style: TextStyle(fontSize: 14),
              ),
              Spacer(),
              if (numTagsAdded! > 0)
                Container(
                  margin: EdgeInsets.only(left: 8.0),
                  padding: EdgeInsets.all(4.0),
                  decoration: BoxDecoration(
                    shape: BoxShape.circle,
                    color: Colors.grey,
                  ),
                  child: Text(
                    '$numTagsAdded',
                    style: TextStyle(
                      color: Colors.white,
                      fontSize: 12.0,
                      fontWeight: FontWeight.bold,
                    ),
                  ),
                ),
            ],
          ),
          children: [
            Padding(
              padding: const EdgeInsets.symmetric(horizontal: 15.0),
              child: Text(
                MyLocalizations.of(context, 'enable_auto_hashtag_text'),
                style: TextStyle(fontSize: 11, color: Colors.grey[600]),
              ),
            ),
            StringMultilineTags(updateTagsNum: updateTagsNum),
          ],
        ),
      ),
    ],
  );
}

Widget _buildSupportSection(BuildContext context) {
  return Column(
    crossAxisAlignment: CrossAxisAlignment.start,
    children: [
      Text("SUPPORT (Hardcoded)"),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'info_scores_txt'),
        () => _navigateToWebview(context, 'url_scoring_1'),
      ),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'about_the_project_txt'),
        () => _navigateToWebview(context, 'url_about_project'),
      ),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'partners_txt'),
        () => Navigator.push(
          context,
          MaterialPageRoute(builder: (context) => PartnersPage()),
        ),
      ),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'coordination_txt'),
        () => _navigateToWebview(context, 'url_about_us'),
      ),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'mailing_mosquito_samples'),
        () => Navigator.push(
          context,
          MaterialPageRoute(builder: (context) => CampaignTutorialPage()),
        ),
        addDivider: false,
      ),
    ],
  );
}

Widget _buildLegalSection(BuildContext context) {
  return Column(
    crossAxisAlignment: CrossAxisAlignment.start,
    children: [
      Text("LEGAL (Hardcoded)"),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'terms_of_use_txt'),
        () => _navigateToWebview(context, 'terms_link'),
      ),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'privacy_txt'),
        () => _navigateToWebview(context, 'privacy_link'),
      ),
      SettingsMenuWidget(
        MyLocalizations.of(context, 'license_txt'),
        () => _navigateToWebview(context, 'lisence_link'),
      ),
    ],
  );
}

void _navigateToWebview(BuildContext context, String urlKey) {
  Navigator.push(
    context,
    MaterialPageRoute(
      builder: (context) => InfoPageInWebview(MyLocalizations.of(context, urlKey)),
    ),
  );
}
```

Comment (DigitalSeneca) [reply lib/pages/settings_pages/settings_page.dart:104]: That's right, I agree that would improve readability, I'll change it and get back to you

**Comment by epou** [comment]
BTW, the only thing I miss is the 'UUID' (with copy to clipboard) and "version" information at the bottom of the page (check figma for that).

**Comment by DigitalSeneca** [comment]
I added the UUID and version here. I think it makes sense to address the font style in a different PR because it affects all pages of the app, not only the Account page
image

**Review by epou — APPROVED** [review]

**Pull request merged**